module/zfs/zfs_replay.c

```diff
@@ -667,13 +681,39 @@ zfs_replay_rename(void *arg1, void *arg2, boolean_t byteswap)
 	if (lr->lr_common.lrc_txtype & TX_CI)
 		vflg |= FIGNORECASE;
 
-	error = zfs_rename(sdzp, sname, tdzp, tname, kcred, vflg);
+	error = zfs_rename(sdzp, sname, tdzp, tname, kcred, vflg, rflags);
 
 	zrele(tdzp);
 	zrele(sdzp);
 	return (error);
 }
 
+static int
+zfs_replay_rename(void *arg1, void *arg2, boolean_t byteswap)
+{
+	return (do_zfs_replay_rename(arg1, arg2, byteswap, 0));
+}
+
+static int
+zfs_replay_rename_exchange(void *arg1, void *arg2, boolean_t byteswap)
+{
+#ifdef __linux__
+	return (do_zfs_replay_rename(arg1, arg2, byteswap, RENAME_EXCHANGE));
+#else
+	return (SET_ERROR(ENOTSUP));
```

Comment:
I think we should panic here. Either the feature is active on disk and we imported it because our software supports it, or the feature is not active on disk and therefore there wouldn't be any log records of this type.